tests/integration/test_hyper_transformer.py

```diff
@@ -83,110 +85,60 @@ def _reverse_transform(self, data):
 
     return data
 
 
-def get_reversed(index=None):
-    reverse_transformed = get_input_data_without_nan(index)
-    reverse_transformed['bool'] = reverse_transformed['bool'].astype('O')
-    return reverse_transformed
-
-
-def get_transformed_data(index=None):
-    transformed = pd.DataFrame({
-        'integer.value': [1, 2, 1, 3],
-        'float.value': [0.1, 0.2, 0.1, 0.1],
-        'categorical.value': [0.375, 0.375, 0.875, 0.375],
-        'bool.value': [0.0, 0.0, 1.0, 0.0],
-        'datetime.value': [
-            1.2649824e+18,
-            1.262304e+18,
-            1.2649824e+18,
-            1.2649824e+18
-        ],
-        'names.value': [0.375, 0.875, 0.375, 0.375]
-    })
-    if index:
-        transformed.index = index
-
-    return transformed
-
-
-def get_transformed_nan_data():
+def get_transformed_data():
+    index = [4, 6, 3, 8, 'a', 1.0, 2.0, 3.0]
+    datetimes = [
+        1.263069e+18,
+        1.264982e+18,
+        1.262304e+18,
+        1.262304e+18,
+        1.262304e+18,
+        1.264982e+18,
+        1.262304e+18,
+        1.262304e+18
+    ]
     return pd.DataFrame({
-        'integer': [1, 2, 1, 3, 1],
-        'float': [0.1, 0.2, 0.1, 0.125, 0.1],
-        'float#1': [0.0, 0.0, 0.0, 1.0, 0.0],
-        'categorical': [0.3, 0.3, 0.9, 0.7, 0.3],
-        'bool': [0.0, -1.0, 0.0, 1.0, 0.0],
-        'bool#1': [0.0, 1.0, 0.0, 0.0, 0.0],
-        'datetime': [
-            1.2636432e+18, 1.2649824e+18, 1.262304e+18,
-            1.2649824e+18, 1.262304e+18
-        ],
-        'datetime#1': [1.0, 0.0, 0.0, 0.0, 0.0],
-        'names': [0.3, 0.8, 0.8, 0.3, 0.3],
-    })
+        'integer.value': [1, 2, 1, 3, 1, 4, 2, 3],
+        'float.value': [0.1, 0.2, 0.1, 0.2, 0.1, 0.4, 0.2, 0.3],
+        'float.is_null': [0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0],
+        'categorical.value': [0.3125, 0.3125, 0.9375, 0.75, 0.3125, 0.75, 0.3125, 0.3125],
+        'bool.value': [0.0, -1.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0],
+        'bool.is_null': [0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+        'datetime.value': datetimes,
+        'datetime.is_null': [1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+        'names.value': [0.3125, 0.75, 0.75, 0.3125, 0.3125, 0.9375, 0.3125, 0.3125]
+    }, index=index)
 
 
-def get_transformers():
-    return {
-        'integer': {
-            'class': 'NumericalTransformer',
-            'kwargs': {
-                'dtype': np.int64,
-            }
-        },
-        'float': {
-            'class': 'NumericalTransformer',
-            'kwargs': {
-                'dtype': np.float64,
-            }
-        },
-        'categorical': {
-            'class': 'CategoricalTransformer'
-        },
-        'bool': {
-            'class': 'BooleanTransformer'
-        },
-        'datetime': {
-            'class': 'DatetimeTransformer'
-        },
-        'names': {
-            'class': 'CategoricalTransformer',
-        },
-    }
+DETERMINISTIC_CATEGORICAL = deepcopy(DEFAULT_TRANSFORMERS)
```

Comment:
minor: could we make this more explicit. Perhaps `DEFAULTS_WITH_DETERMINISTIC_CATEGORICAL`